File: LeetCode/121. Best Time to Buy and Sell Stock/solution.go
func maxProfit(prices []int) int {
	r := 0
	result := 0

	for i := 0; i < len(prices); i++ {
		if prices[i] < prices[r] {
			r = i
		}
		result = max(result, prices[i]-prices[r])
	}
	return result
}

File: LeetCode/121. Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        res = 0
        for right in range(len(prices)):
            if prices[right] < prices[left]:
                left = right
            res = max(res, (prices[right] - prices[left]))
        return res    